### PR TITLE
feat(version): added auto update method

### LINKED ISSUE #24: [Contracts] Auto-update version number during update cycle
As a developer, I want to automatically update the contract version number after every deployment

### PR DESCRIPTION
**Pull request by bitbeckers**
* changes version() from pure to view
* add method callable by upgrader role to increment version number with 1

fixes #24

### CODE REVIEW

File: test/setup.ts [hunk 1]
```diff
@@ -58,8 +58,8 @@ const setupTest = deployments.createFixture<
 });
 
 export const setupImpactScopes = async (
-  contract: HypercertMinterV0,
-  contractAtAddress?: HypercertMinterV0,
+  contract: HypercertMinterV1 | HypercertMinterV0,
```
Comment (moconnell) [inline test/setup.ts:61]: Hmm this looks a bit of a PITA every time we do an upgrade - maybe worth defining a union type somewhere and then we just have one place to update?

File: test/setup.ts [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-  minter: HypercertMinterV0;
+  minter: HypercertMinterV1;
 };
 
 export type HypercertCollection = {
-  minter: HypercertMinterV0;
-  deployer: AddressedHypercertMinterV0;
-  user: AddressedHypercertMinterV0;
-  anon: AddressedHypercertMinterV0;
+  minter: HypercertMinterV1;
```
Comment (moconnell) [inline test/setup.ts:13]: Perhaps here also as below worth defining a "pointer" e.g. HypercertMinterCurrentVersion or somesuch and then you just have one place to update..?

File: test/setup.ts [hunk 3]
```diff
@@ -34,13 +34,13 @@ const setupTest = deployments.createFixture<
   const { deployer, user, anon } = await getNamedAccounts();
 
   // Contracts
-  const minter: HypercertMinterV0 = await ethers.getContract("HypercertMinterV0");
+  const minter: HypercertMinterV1 = await ethers.getContract(HypercertMinter_V1);
```
Comment (moconnell) [inline test/setup.ts:37]: Same thing - define a type that points to the current version and then there is only one line to update each time?

File: test/setup.ts [hunk 4]
```diff
@@ -34,13 +34,13 @@ const setupTest = deployments.createFixture<
...
 
   // Account config
   const setupAddress = async (address: string) => {
     return {
       address: address,
-      minter: <HypercertMinterV0>await ethers.getContract("HypercertMinterV0", address),
+      minter: <HypercertMinterV1>await ethers.getContract(HypercertMinter_V1, address),
```
Comment (moconnell) [inline test/setup.ts:43]: Ditto

**Review by moconnell — CHANGES_REQUESTED** [review]

File: test/setup.ts (hunk 3 above)
Comment (bitbeckers) [reply test/setup.ts:37]: done

**Pull request merged**